File: SpringBootApp/src/main/java/com/example/springbootapp/data/dto/WishlistDto.java
package com.example.springbootapp.data.dto;

import com.example.springbootapp.data.entities.Enums.WishlistVisibility;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@NoArgsConstructor
public class WishlistDto {
    private String id;
    private WishlistVisibility visibility;
    private List<WishlistItemDto> items;
    private String ownerUsername;
}

File: ClientUser/app/src/main/java/com/example/clientuser/model/dto/WishListDto.kt
package com.example.clientuser.model.dto

import com.example.clientuser.model.enumerator.WishListVisibility
import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = true)
data class WishListDto(
    val id: String,
    val ownerUsername: String,
    val items: List<WishlistItemDto>,
    val visibility: WishListVisibility
)
